7.0 add account journal entry posted async lmi

lmignon opened the pull request:
This module add a new checkbox on the journal to post asynchronously entries
created for the journal without any manual validation.

In this module we take advantage of the queue and channel system from Odoo
Connector to sequentially post entries in background and therefore avoid 
concurrent updates on ir_sequence when the system is intensely used.

To work as expected this PR depends of https://github.com/OCA/connector/pull/53

lmignon committed:
[ADD] new module account_journal_entry_posted_async

lmignon committed:
[IMP] Improve documentation

sbidoul commented on account_journal_entry_posted_async/models/account_journal.py:
```diff
@@ -0,0 +1,76 @@
+from openerp import exceptions
+from openerp.osv import orm, fields
+
+SYNC_OR_ASYNC_ERR_MSG = _('You must choose between sync or async post '
```
I think you can't use _() here because you have no translation context at import time

lmignon replied:
You're right, I fix

lmignon committed:
[FIX] no translation context available at import time

sbidoul commented on account_journal_entry_posted_async/models/account_journal.py:
```diff
@@ -0,0 +1,76 @@
+    _columns = {
+        'entry_posted_async': fields.boolean(
+            'Asynchronously Post for Manual Entries',
+            help='Check this box if you don\'t want new journal entries to '
```
if you _do_ want?

lmignon committed:
[FIX] s/don't want/want

sbidoul commented on account_journal_entry_posted_async/models/account_journal.py:
```diff
@@ -0,0 +1,76 @@
+        res = {'value': value}
+        if boolval:
+            value['entry_posted'] = False
+        return res
```
Would it work too by forcing entry_posted to true when entry_posted_async is true and then overriding button_validate to work asynchronously?

lmignon replied:
@sbidoul I'll investigate your proposal. The code could be simplified by overriding the button_validate.

lmignon committed:
[IMP] Simplification of the code
By overriding the button_validate function in the account_move class we no more need to override the function 'create' in accoun_move and account_move_line and duplicate the logic behind the field 'entry_posted'. Moreover, the module no more depends of account_move_batch_validate.

lmignon committed:
[FIX] correctly call super

laetitia-gangloff commented:
:+1: (functional use)

lmignon commented:
branch rebased and all tests are green.
@sbidoul IMO this one is ready for review and the milestone should be 7.0

guewen commented on account_journal_entry_posted_async/models/account_move.py:
```diff
@@ -0,0 +1,69 @@
+        session = ConnectorSession(cr, uid, context=context)
+        for move in self.browse(cr, uid, ids, context=context):
+            if not move.journal_id.entry_posted_async:
+                self.do_button_validate(cr, uid, ids, context)
```
Nitpicking: `context=context`

guewen commented:
Wow! <3
:+1:

sbidoul commented on account_journal_entry_posted_async/__openerp__.py:
```diff
@@ -0,0 +1,68 @@
+<http://odoo-connector.com/guides/jobrunner.html>`_ to see how to configure
+the capacity of channels .
+
+    """,
```
Remove description.

lmignon committed:
[FIX] Always pass the context as kwargs

laetitia-gangloff commented:
2 thumb up and more than 5 days, is it not mergeable ?
